src/components/details-todo-content.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback } from 'react';
+import * as S from '../styles/styled';
+import { useHistory, useParams } from 'react-router-dom';
+import { ToDoCard } from './todo-card';
+import { getToDo } from 'services/get-todo-service';
+
+export function DetailsTodoContent() {
+  const history = useHistory();
+  const { id }: { id: string } = useParams();
+  const [toDo, setToDo] = React.useState<{
+    id: number;
+    title: string;
+    description: string;
+  }>({
+    id: 0,
+    title: '',
+    description: '',
+  });
+
+  const todoById = useCallback(async () => {
+    const response = await getToDo(Number(id));
+
+    setToDo(response);
+  }, [id]);
+
+  React.useEffect(() => {
+    todoById();
+  }, [todoById]);
```

Comment:
aqui poderia ser apenas isso que não faria muita diferença, desde que ficasse pouco código no effect:
```suggestion
  useEffect(() => {
    const fetchTodos = async () => {
      const response = await getToDo(Number(id));
      setToDo(response);
    };
    fetchTodos();
  }, []);
```
ou até assim, menos bonito mas mais simples:
```suggestion
  useEffect(() => {
    getToDo().then(setToDo);
  }, []);
```
prefiro a primeira sugestão